code/modules/projectiles/guns/flamer.dm

```diff
@@ -30,6 +30,40 @@
 	fire_delay = 4
 
 
+/obj/item/weapon/gun/flamer/mini_flamer
+	name = "mini flamethrower"
+	desc = "A weapon-mounted refillable flamethrower attachment.\nIt is designed for short bursts."
+	icon = 'icons/Marine/marine-weapons.dmi'
+	icon_state = "flamethrower"
+	
+	flags_gun_features = GUN_UNUSUAL_DESIGN|GUN_AMMO_COUNTER|GUN_WIELDED_FIRING_ONLY|GUN_WIELDED_STABLE_FIRING_ONLY|GUN_IS_ATTACHMENT|GUN_ATTACHMENT_FIRE_ONLY
```

Comment:
why GUN_AMMO_COUNTER ? can you see the ammo?

Reply:
yes